rust/sealevel/client/src/main.rs

```diff
@@ -345,6 +345,39 @@ struct TransferOwnership {
     new_owner: Pubkey,
 }
 
+#[derive(Args)]
+struct Igp {
+    #[arg(long, short, default_value_t = HYPERLANE_TOKEN_PROG_ID)]
+    program_id: Pubkey,
+    #[command(subcommand)]
+    cmd: GetSetCmd<GetIgpArgs, SetIgpArgs>,
+}
+
+#[derive(Subcommand)]
+enum GetSetCmd<G: Args, S: Args> {
+    Get(G),
+    Set(S),
+}
+
+#[derive(Args)]
+struct SetIgpArgs {
+    igp_program: Pubkey,
+    #[arg(value_enum)]
+    igp_type: IgpType,
+    igp_account: Pubkey,
+}
+
+#[derive(Args)]
+struct GetIgpArgs {
+    token_type: TokenType,
+}
+
+#[derive(ValueEnum, Clone)]
+enum IgpType {
```

Comment:
I take it there's no way to use the existing `InterchainGasPaymasterType` enum w/ clap?

Reply:
None that I could find. It's probably possible to auto-generate this type with a macro but we can't just drop the inner enum types of `InterchainGasPaymasterType` for clap